### PR TITLE
[task-25] Migrate tally-frontend from JavaScript to TypeScript

### PR DESCRIPTION
**Pull request by michello**
## Description

Added changes for JavaScript to TypeScript migration, fixed errors involved with incorrect typing, and fixed tests that failed as a result from type fixes.

Changes specific for JavaScript to TypeScript migration involve:
* Renaming file extension from .js to .tsx or ts
* Adding typing to parameters
* Adding babel config
* Installing @babel-typescript and ts-jest

Changes specific for fixing errors involve:
* Removing class prop from several existing components
* Fixing the way we set Tax and Tip, as we were setting a String value when it should be Number.

## How Has This Been Tested?
* Unit tests succeed
* [Screen recording](https://user-images.githubusercontent.com/14842270/164352525-2a5a6b9f-2461-4186-826e-f4ac974d99c0.mov)

### CODE REVIEW

File: src/tests/components/Tip.test.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ describe('Tip component', () => {
     jest.clearAllMocks();
   });
 
-  const calculateTip = (perecentage) => (perecentage * total).toFixed(2);
+  const calculateTip = (perecentage) => parseFloat((perecentage * total).toFixed(2));
```
Comment (calvinywcheng) [inline src/tests/components/Tip.test.tsx:20]:
```suggestion
  const calculateTip = (percentage) => parseFloat((perecentage * total).toFixed(2));
```
also, should this be type annotated?

File: src/tests/pages/App.test.tsx [hunk 2]
```diff
@@ -21,4 +21,4 @@ describe('App page', () => {
     expect(wrapper.find(AddPartyMember)).toHaveLength(1);
     expect(wrapper.find(Tip)).toHaveLength(1);
   });
-});
+});
```
Comment (calvinywcheng) [inline src/tests/pages/App.test.tsx:24]:
```suggestion
});

```
files should typically end with new line
https://stackoverflow.com/q/729692

File: src/components/Tip.tsx [hunk 3]
```diff
@@ -55,7 +52,7 @@ export default function Tip({ total, setTip }) {
             onChangeText={(tipPercentageInput) => {
               const tipPercentageInputInt = parseInt(tipPercentageInput)
               if (isTipValid(tipPercentageInputInt)) {
-                changePercentage(total, tipPercentageInputInt, true);
+                changeTipValues(total, tipPercentageInputInt)
```
Comment (calvinywcheng) [inline src/components/Tip.tsx:55]: does this need to also set the custom tip to true?

Comment (michello) [reply src/components/Tip.tsx:55]: No, this <TextInput/> component would not be visible if custom tip is not already set to `True` (see L48). Therefore, at this point, custom tip is already set to true.

File: src/tests/components/Tip.test.tsx (hunk 1 above)
Comment (michello) [reply src/tests/components/Tip.test.tsx:20]: If I must...

**Review by calvinywcheng — APPROVED** [review]

**Review by rosyang — APPROVED** [review]
Looks good! Thanks for doing the migration work~

File: src/components/Tax.tsx [hunk 4]
```diff
@@ -3,18 +3,25 @@ import PropTypes from 'prop-types';
...
-    <View class="tax-amount">
+    <View>
         <Text>Enter tax here:</Text>
         <TextInput
+          placeholder="0"
           onChangeText={(taxAmountInput) => {
             if (isTaxAmountValid(taxAmountInput)) {
-              setTax(parseFloat(taxAmountInput).toFixed(2));
+              const taxAmount = roundToNearestCent(taxAmountInput)
+              // parseFloat(parseFloat(taxAmountInput).toFixed(2))
```
Comment (elliely32) [inline src/components/Tax.tsx:23]: nitpick: remove comment
(nice function lmaoo)

**Pull request merged**